include/mbedtls/ecdh.h

```diff
@@ -36,6 +36,15 @@
 
 #include "ecp.h"
 
+/**
+ * Use a backward compatible ECDH context.
+ *
+ * If a feature disables this flag, the corresponding description in config.h
+ * must contain a warning stating that the feature breaks backward
+ * compatibility.
+ */
+#define MBEDTLS_ECDH_LEGACY_CONTEXT
```

Comment:
This makes it impossible to switch off the legacy context without editing the file.

This flag belongs in `config.h`.

Reply:
This flag is not a standalone feature at this point. The new context will be used by other features, like the upcoming alternative implementation. Those are features that need a flag in `config.h` and need to conditionally turn off the legacy flag here to make use of the new context.

Reply:
But in the meantime, we can't even test a build with `MBEDTLS_ECDH_LEGACY_CONTEXT` not defined.

Reply:
Yes, it can only be tested manually as it is. This is a problem, but only until the first feature that uses this option. After that it will be tested whenever the corresponding feature is tested.

If this is not enough, we can extend `config.pl` with new commands (eg setmod/unsetmod) to set/unset flags in module headers to enable automatic testing.

Reply:
Would it help if we merged our changes with this PR so that there is at least one feature that uses the new macros? Also, I think it would be possible to base our PR on the branch of this PR, so that they can be merged in tandem.

Reply:
I feel that it would be cleaner to handle the two PRs separately for now.

Reply:
No changes are necessary to us `config.pl` on this. You can just do this:
```
./scripts/config.pl unset MBEDTLS_ECDH_LEGACY_CONTEXT --file include/mbedtls/ecdh.h
```
That will comment out the configuration.